octobot_trading/exchange_data/ohlcv/preloaded_candles_manager.py

```diff
@@ -0,0 +1,114 @@
+#  Drakkar-Software OctoBot-Trading
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+import numpy as np
+
+import octobot_commons.enums as enums
+import octobot_commons.constants as commons_constants
+
+import octobot_trading.exchange_data.ohlcv.candles_manager as candles_manager
+
+
+class PreloadedCandlesManager(candles_manager.CandlesManager):
+
+    def get_preloaded_symbol_candles_count(self):
+        return len(self.time_candles)
+
+    def get_preloaded_symbol_close_candles(self):
+        return self.close_candles
+
+    def get_preloaded_symbol_open_candles(self):
+        return self.open_candles
+
+    def get_preloaded_symbol_high_candles(self):
+        return self.high_candles
+
+    def get_preloaded_symbol_low_candles(self):
+        return self.low_candles
+
+    def get_preloaded_symbol_time_candles(self):
+        return self.time_candles
+
+    def get_preloaded_symbol_volume_candles(self):
+        return self.volume_candles
+
+    def _set_all_candles(self, new_candles_data):
+        self.close_candles = self._get_candle_values_array(new_candles_data, enums.PriceIndexes.IND_PRICE_CLOSE.value)
+        self.open_candles = self._get_candle_values_array(new_candles_data, enums.PriceIndexes.IND_PRICE_OPEN.value)
+        self.high_candles = self._get_candle_values_array(new_candles_data, enums.PriceIndexes.IND_PRICE_HIGH.value)
+        self.low_candles = self._get_candle_values_array(new_candles_data, enums.PriceIndexes.IND_PRICE_LOW.value)
+        self.time_candles = self._get_candle_values_array(new_candles_data, enums.PriceIndexes.IND_PRICE_TIME.value)
+        self.volume_candles = self._get_candle_values_array(new_candles_data, enums.PriceIndexes.IND_PRICE_VOL.value)
+
+    def _get_candle_values_array(self, candles, key):
+        return np.array([candle[key] for candle in candles])
+
+    def _get_candle_index(self, candle):
```

Comment:
Could we add more documentation in this method please? I don't understand how it works

Reply:
done